.github/docker/Dockerfile.musl

```diff
@@ -0,0 +1,18 @@
+ARG PLATFORM=arm64
+ARG NODE_VERSION=16.20.1
+
+FROM ${PLATFORM}/node:${NODE_VERSION}-alpine AS build
+
+WORKDIR /mongodb-client-encryption
+COPY . .
+
+RUN apk --no-cache add make g++ libc-dev curl bash python3 py3-pip vim cmake git
```

Comment:
Is vim a requirement to build? 😅

Reply:
No but its a requirement if you work in the container

This must have slipped through the cracks in zstd and then got copied here.  I don't feel strongly about leaving it vs removing it

Reply:
Everything adds to the time it takes for our tests/builds to finish, I can throw up a quick PR to remove it from zstd too so we're not out of sync

Reply:
https://github.com/mongodb-js/zstd/pull/70

Reply:
I guess, but I think this is a micro-optimization.  Installing vim takes a fraction of the total build + test time, which is dominated by building deps, installing deps (zstd/libmongocrypt) and running the tests

If we really want these to run faster, we should publish docker images that come with dependencies already installed, so all we need to do in CI is copy in the files and build the bindings